### PR TITLE
Added history option to list

### PR DESCRIPTION
**Pull request by breillyrh**

### CODE REVIEW

File: bin/dock-pulp.py [hunk 1]
```diff
@@ -821,19 +823,40 @@ def do_list(bopts, bargs):
                     except KeyError:
                         output[layer] = {}
 
-                    output[layer][manifest] = repo['manifests'][manifest]['tag']
+                    try:
```
Comment (lcarva) [inline bin/dock-pulp.py]: FYI, a more pythonic way of doing this:

```
output.setdefault('layer', {})
output['layer'].setdefault('manifest', {})
```

The above also creates the initial 'layer' key if needed (which is right above where it lets me comment)

File: dockpulp/__init__.py [hunk 2]
```diff
@@ -743,6 +744,31 @@ def listRepos(self, repos=None, content=False):
                     r['manifests'][manifest['metadata']['digest']] = {}
                     r['manifests'][manifest['metadata']['digest']]['tag'] = manifest['metadata']['tag']
                     r['manifests'][manifest['metadata']['digest']]['layers'] = layers
+                if history:
+                    for manifest in r['manifests'].keys():
+                        data = self._get('/pulp/docker/v2/%s/manifests/%s' % (blob['id'], manifest))
+                        
+                        # Unsure if all v2 images will have v1 history
```
Comment (twaugh) [inline dockpulp/__init__.py]: Pretty sure they always will, but it's no bad thing to code defensively when handling docker things. :-)

**Comment by twaugh** [comment]
LGTM

File: bin/dock-pulp.py [hunk 3]
```diff
@@ -821,19 +823,40 @@ def do_list(bopts, bargs):
...
                 images = output.keys()
                 for image in images:
                     log.info('')
                     manifests = output[image].keys()
+                    tagoutput = []
                     for manifest in manifests:
-                        log.info('  Digest: %s  Tag: %s' %
-                            (manifest, output[image][manifest]))
-                    log.info('    Layers: ')
+                        log.info('  Manifest: %s  Tag: %s' %
```
Comment (lcarva) [inline bin/dock-pulp.py]: The preferred way is to pass each replacement as an arg to logging function:

```
log.info('  Manifest: %s  Tag: %s', manifest, output[image][manifest])
```

**Comment by lcarva** [comment]
Tiny nitpicking comments, otherwise LGTM

File: bin/dock-pulp.py (hunk 1 above)
Comment (breillyrh) [inline bin/dock-pulp.py]: Aha, that's much cleaner.

**Pull request merged**